install/kubernetes/cilium/templates/cilium-ingress-service.yaml

```diff
@@ -28,6 +28,9 @@ spec:
   {{- if .Values.ingressController.service.loadBalancerClass }}
   loadBalancerClass: {{ .Values.ingressController.service.loadBalancerClass }}
   {{- end }}
+  {{- if hasKey .Values.ingressController.service "allocateLoadBalancerNodePorts" }}
```

Comment:
Won't this always be true if we have a default `~` value here? 

We should use `{{- if (not (kindIs "invalid" .Values.ingressController.service.allocateLoadBalancerNodePorts)) }}` instead here (c.f. `bpf.hostLegacyRouting` for a similar example).

Reply:
I agree.

Reply:
thanks for your pointer both :100: